flink-core/src/main/java/org/apache/flink/api/connector/source/lib/SourceReaderFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.connector.source.lib;
+
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.SourceReaderContext;
+import org.apache.flink.api.connector.source.SourceSplit;
+
+import java.io.Serializable;
+
+/**
+ * A factory for creating source reader instances.
+ *
+ * @param <OUT> The type of the output elements.
+ */
+public interface SourceReaderFactory<OUT, SplitT extends SourceSplit> extends Serializable {
```

Comment:
Would it make sense to have the Source interface extend this interface to keep the signatures the same?

Reply:
Do I get it correctly that the idea is to change:
`public interface Source<T, SplitT extends SourceSplit, EnumChkT> extends Serializable`
to
`public interface Source<T, SplitT extends SourceSplit, EnumChkT> extends SourceReaderFactory<T, SplitT>`
and to pull up the `createReader` into the `SourceReaderFactory` instead of the current `newSourceReader`?

Reply:
yes

Reply:
Done.